File: windows/Zikr/ZikrData.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Web.Script.Serialization;

namespace Zikr
{
    /// <summary>
    /// Loads the bundled zikr list. Same 21 general adhkar as the macOS
    /// build's ZikrList.swift and the Linux build's data/zikr.json (kept
    /// in sync by hand).
    /// </summary>
    public static class ZikrData
    {
        private static readonly Lazy<List<ZikrItem>> _all = new Lazy<List<ZikrItem>>(Load);

        public static List<ZikrItem> All => _all.Value;

        private static readonly Random _rng = new Random();

        public static ZikrItem RandomZikr()
        {
            var all = All;
            return all[_rng.Next(all.Count)];
        }

        private static List<ZikrItem> Load()
        {
            string path = ResolveDataPath();
            string json = File.ReadAllText(path, System.Text.Encoding.UTF8);
            var serializer = new JavaScriptSerializer();
            var raw = serializer.Deserialize<List<Dictionary<string, object>>>(json);

            return raw.Select(d => new ZikrItem
            {
                Id = Convert.ToInt32(d["id"]),
                Arabic = (string)d["arabic"],
                Transliteration = (string)d["transliteration"],
                Translation = (string)d["translation"],
            }).ToList();
        }

        private static string ResolveDataPath()
        {
            string exeDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
            string candidate = Path.Combine(exeDir ?? ".", "Resources", "zikr.json");
            if (File.Exists(candidate)) return candidate;

            candidate = Path.Combine(exeDir ?? ".", "zikr.json");
            if (File.Exists(candidate)) return candidate;

            throw new FileNotFoundException("zikr.json not found next to the executable");
        }
    }
}

File: android/app/src/main/java/com/abu/zikr/data/ZikrData.kt
package com.abu.zikr.data

import android.content.Context
import com.abu.zikr.R
import org.json.JSONArray

/**
 * Loads the bundled zikr list. Same 21 general adhkar as the other
 * builds' ZikrList.swift/zikr_data.py/ZikrData.cs - res/raw/zikr.json
 * is kept in sync with them by hand.
 */
object ZikrData {
    @Volatile private var cached: List<ZikrItem>? = null

    fun all(context: Context): List<ZikrItem> {
        cached?.let { return it }
        synchronized(this) {
            cached?.let { return it }
            val text = context.resources.openRawResource(R.raw.zikr)
                .bufferedReader()
                .use { it.readText() }
            val loaded = parse(text)
            cached = loaded
            return loaded
        }
    }

    /** Pure function so parsing is unit-testable without a Context. */
    fun parse(jsonText: String): List<ZikrItem> {
        val array = JSONArray(jsonText)
        return (0 until array.length()).map { i ->
            val obj = array.getJSONObject(i)
            ZikrItem(
                id = obj.getInt("id"),
                arabic = obj.getString("arabic"),
                transliteration = obj.getString("transliteration"),
                translation = obj.getString("translation"),
            )
        }
    }

    fun random(context: Context): ZikrItem = all(context).random()
}
